src/python/pants/core/goals/generate_lockfiles.py

```diff
@@ -124,6 +125,19 @@ class RequestedUserResolveNames(Collection[str]):
     """
 
 
+@union(in_scope_types=[EnvironmentName])
+@dataclass(frozen=True)
+class LockfileGenerated:
+    """Hook for post-processing after lockfile generation."""
+
+    result: GenerateLockfileResult
+
+
+@dataclass(frozen=True)
+class LockfileGeneratedPostProcessing:
+    process: InteractiveProcess | None = None
```

Comment:
So, unless we intend users to extend this point, I don't think that this should be a `@union`.

The things that a user can safely do at this point are pretty limited: you can essentially only execute readonly operations on the lockfile... otherwise you would invalidate the header, and we wouldn't be able to tell that we needed to regenerate it.

If #17347 adds an option (or default!) to render the diff after regenerating the lockfile, that doesn't need to be behind a `@union`: if after using that feature a user wants to extend this point and add more ways to render the diff, then maybe we could add an extension point... otherwise it feels like premature generalization.

Reply:
@stuhood are you suggesting to move the `lockfile_diff` subsystem to `pants/core/subsystems`? (if not moving, I don't see how to avoid a dependency from `core/goals` to the `project_info` backend without the `@union`)

Reply:
I'm suggesting that #17347 should be built in rather than enabled by an experimental backend, and enabled by a non-default flag (possibly named `--experimental-*` if you think that it is risky in some way).

Reply:
not very risky, but maybe incomplete, as I'm not sure if the lockfile-diff tool will process every lockfile thrown at it yet ;)